arch/arm/Kconfig

```diff
@@ -344,11 +345,11 @@ config ARCH_CHIP_RP2040
 	select ARCH_CORTEXM0
 	select ARCH_HAVE_RAMVECTORS
 	select ARCH_HAVE_MULTICPU
-	select ARCH_HAVE_TESTSET
 	select ARCH_HAVE_I2CRESET
 	select ARM_HAVE_WFE_SEV
 	select ARCH_HAVE_PWM_MULTICHAN
 	select ARCH_BOARD_COMMON
+	select ARCH_HAVE_CUSTOM_TESTSET
```

Comment:
Add ARCH_HAVE_CUSTOM_TESTSET to the deconfig of the corresponding architecture.

Reply:
> Add ARCH_HAVE_CUSTOM_TESTSET to the deconfig of the corresponding architecture.

in NuttX, there are more than 100 defconfigs using `CONFIG_ARCH_CHIP_RP2040` in total. If we were to modify the defconfigs, we could have to modify over 100 defconfig files. for example the following boards defconfig all need to modify:
```
boards/arm/rp2040/raspberrypi-pico-w/configs/telnet/defconfig:19:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/smp/defconfig:18:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/usbmsc/defconfig:21:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/audiopack/defconfig:25:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/ssd1306/defconfig:21:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/composite/defconfig:21:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/displaypack/defconfig:24:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/waveshare-lcd-1.3/defconfig:23:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/st7735/defconfig:18:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/nsh-flash/defconfig:17:CONFIG_ARCH_CHIP_RP2040=y
boards/arm/rp2040/raspberrypi-pico-w/configs/lcd1602/defconfig:18:CONFIG_ARCH_CHIP_RP2040=y
...
```